sdk/java-sdk/src/main/java/kalix/javasdk/action/Action.java

```diff
@@ -61,6 +64,14 @@ public final Effect.Builder effects() {
     return ActionEffectImpl.builder();
   }
 
+  /** Returns a {@link TimerScheduler} that can be used to schedule further in time. */
+  public final TimerScheduler timers() {
+    ActionContextImpl impl =
+        (ActionContextImpl)
+            actionContext("Timers can only be scheduled or cancelled when handling a message.");
+    return new TimerSchedulerImpl(impl.anySupport(), impl.system());
```

Comment:
As I said earlier, I think we should create it upfront so that it is always available from the ActionContextImpl, we can then still make the public API for the user this method that delegates to it. That way we can easily inject a stateful pretend-timer in the testkit.

But we can ofc also do that refactoring once we work on testkit support.